mapper/orm/src/main/java/org/hibernate/search/mapper/orm/search/loading/impl/HibernateOrmNonEntityIdPropertyEntityLoader.java

```diff
@@ -113,13 +114,16 @@ public void loadBlocking(List<EntityReference> references, Map<? super EntityRef
 	}
 
 	@SuppressWarnings("unchecked") // Cast is safe because entityPersister represents type E. See Factory.doCreate().
-	private Query<? extends E> createQuery(int fetchSize) {
+	private Query<? extends E> createQuery(int fetchSize, Integer timeout) {
 		Query<? extends E> query = (Query<? extends E>) HibernateOrmUtils.createQueryForLoadByUniqueProperty(
 				session, entityPersister, documentIdSourcePropertyName,
 				DOCUMENT_ID_SOURCE_PROPERTY_PARAMETER_NAME
 		);
 
 		query.setFetchSize( fetchSize );
+		if ( timeout != null ) {
+			query.setTimeout( HibernateOrmComposableEntityLoader.getTimeoutInSeconds( timeout ) );
```

Comment:
Rounding up the timeout to the next second is definitely not acceptable for search queries. People need sub-second response times for the overwhelming majority of use cases.

Use `query.setHint( QueryHints.SPEC_HINT_TIMEOUT, <value in milliseconds> )` instead.

Reply:
Right! I forgot we have this option

Reply:
Anyway I'm not sure that with `QueryHints.SPEC_HINT_TIMEOUT` we can have more precision. See `AbstractProducedQuery#setHint`, in particular the if branch when `SPEC_HINT_TIMEOUT.equals( hintName )`.

Reply:
Probably it is better to use `HINT_TIMEOUT`.

Reply:
> Anyway I'm not sure that with QueryHints.SPEC_HINT_TIMEOUT we can have more precision. See AbstractProducedQuery#setHint, in particular the if branch when SPEC_HINT_TIMEOUT.equals( hintName ).

Well, that's underwhelming... Looks like JDBC only allows second precision. I guess at least we'll be ready for a potential future version of the spec, if they ever have a look at that...

What's more annoying is how are we going to test this... We need a query that takes significantly more than one second to execute, or we need to mock the `TimingSource` (which has its own disadvantage, as I mentioned over the chat). Let's discuss this on Zulip if you need to.